todo.html

```diff
@@ -0,0 +1,111 @@
+<!DOCTYPE html>
+<html>
+<head>
+<script type="text/javascript">
+</script>
+</head>
+
+<body>
+<h1>Todo List</h1>
+    <div id="todoList"></div>
+<hr>
+    <input type="text" id="todoDescription" placeholder="Add a todo here"/>
+    <button id="addTodo">Add</button>
+    
+    <div>Remaining tasks:<a id="remainingCount">0</a></div>
+    <div>Completed tasks:<a id="completedCount">0</a></div>
+<script >
+
+var totalTasks = 0;
+
+function addToList(todoItem, todoDescription) {
+    var newTodo = document.createElement("div");
+    var todoCheckbox = document.createElement("input");
+    todoCheckbox.setAttribute('type', 'checkbox');
+    todoCheckbox.setAttribute('name', 'oneCheckbox');
+    todoCheckbox.setAttribute('onchange','changeHandler()');    
+    newTodo.appendChild(todoCheckbox);
+    newTodo.innerHTML = newTodo.innerHTML + todoDescription + '  ';
+    
+
+    var removeTodoBtn = document.createElement("button");
+    var btnText = document.createTextNode("Remove");
+    removeTodoBtn.appendChild(btnText);
+
+    newTodo.appendChild(removeTodoBtn);
+    todoList.appendChild(newTodo);
```

Comment:
You rely on your "view"/DOM to provide your model/data, which is an anti-pattern. You need to separate them out. This is where MVC comes in. What you have here is effectively a controller, but your model and your view are bundled together. 

The drawback to this is that it's harder to test, visualize, reason about, and refactor. It's harder to test because if you just wanted to unit test your logic around todo manipulation you *have* to have a DOM, there is no option not to have one. Then you are doing an integration test - which is more brittle (you rely on something outside of the logic you wrote, like how the DOM functions). The reason it's harder to visualize and therefore reason about is because I have to look at the HTML and JS side by side to see what is going on. It's of course harder to refactor because if I wanted to move the code to a NodeJS app tomorrow, all my logic is dependent on the browser.